content/en/platform/corda/4.8/open-source/generating-a-node-cordform.md

```diff
@@ -0,0 +1,303 @@
+    }
+    [...]
+}
+```
+
+### Sign CorDapp `.jar` files
+
+The default Cordform behaviour is to deploy CorDapp `.jar` files “as built”.
+
+* Prior to Corda 4.0, all CorDapp `.jar` files were unsigned.
+* As of Corda 4.0, CorDapp `.jar` files created by the gradle `cordapp` plug-in are signed by a Corda development certificate by default.
+
+You can use the Cordform `signing` entry to override and customise the signing of CorDapp `.jar` files.
+Signing a CorDapp enables its contract classes to use signature constraints instead of other types of constraints, such as [Contract Constraints](api-contract-constraints.md).
+
+The signing task may use an external keystore, or create a new one.
+You can use the following parameters in the `signing` entry:
+
+
+* `enabled` - the control flag to enable the signing process. It is set to `false` by default. Set to `true` to enable signing.
+* `all` - if set to `true` (default), all CorDapps inside the `cordapp` sub-directory will be signed. If set to `false`, only the generated Cordapp will be signed.
+* `options` - any relevant parameters of [SignJar ANT task](https://ant.apache.org/manual/Tasks/signjar.html) and [GenKey ANT task](https://ant.apache.org/manual/Tasks/genkey.html). By default the `.jar` file is signed by a Corda development key. You can specify the external keystore can be specified. The minimal list of required options is shown below. For other options, see [SignJar task](https://ant.apache.org/manual/Tasks/signjar.html).
+  * `keystore` - the path to the keystore file. The default setting is `cordadevcakeys.jks`. The keystore is shipped with the plug-in.
+  * `alias` - the alias to sign under. The default value is `cordaintermediateca`.
+  * `storepass` - the keystore password. The default value is `cordacadevpass`.
+  * `keypass` - the private key password, if it is different from the keystore password. The default value is `cordacadevkeypass`.
+  * `storetype` - the keystore type. The default value is `JKS`.
+  * `dname` - the distinguished name for the entity. Only use this option when `generateKeystore` is set to `true` (see below).
+  * `keyalg` - the method to use when generating a name-value pair. The default value is `RSA` because Corda does not support `DSA`. Only use this option when `generateKeystore` is set to `true` (see below).
+* `generateKeystore` - the flag to generate a keystore. The default value is `false`. If set to `true`, an "ad hoc" keystore is created and its key is used instead of the default Corda development key or any external key. The same `options` to specify an external keystore are used to define the newly created keystore. In addition,
+  `dname` and `keyalg` are required. Other options are described in [GenKey task](https://ant.apache.org/manual/Tasks/genkey.html). If the existing keystore is already present, the task will reuse it. However if the file is inside the `build` directory,
+  then it will be deleted when the gradle `clean` task is run.
```

Comment:
This is a fairly chunky bullet point, with an additional requirement. Consider how this is laid out, maybe add indented bullet.